### PR TITLE
Add the release note for C++ client 3.1.0

### PR DESCRIPTION
**Pull request by RobertIndie**
Preview:
image
image
image

### CODE REVIEW

**Review by tisonkun — APPROVED** [review]

File: site2/website-next/release-notes/versioned/client-cpp-3.1.0.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+* [fix][doc] explicitly instruct clone the repo by @tisonkun in https://github.com/apache/pulsar-client-cpp/pull/69
+* [feat] Support expiration for chunked messages by @RobertIndie in https://github.com/apache/pulsar-client-cpp/pull/71
+* Add windows release artifacts by @yaalsn in https://github.com/apache/pulsar-client-cpp/pull/72
+* Support linking static dependencies when building with MSVC by @BewareMyPower in https://github.com/apache/pulsar-client-cpp/pull/73
+* [refactor] Remove useless files and config. by @shibd in https://github.com/apache/pulsar-client-cpp/pull/76
+* Fix CI install deps failed. by @shibd in https://github.com/apache/pulsar-client-cpp/pull/96
+* [fix] Flush no batch message when call producer.flush by @shibd in https://github.com/apache/pulsar-client-cpp/pull/98
+* [Build] Use Windows server 2019 to release Windows artifacts by @BewareMyPower in https://github.com/apache/pulsar-client-cpp/pull/99
+* [fix] Fix consumer doesn't clear incomingMessageQueue during seek by @RobertIndie in https://github.com/apache/pulsar-client-cpp/pull/101
+* Fix doxygen generation by @BewareMyPower in https://github.com/apache/pulsar-client-cpp/pull/103
```
Comment (Anonymitaet) [inline site2/website-next/release-notes/versioned/client-cpp-3.1.0.md:29]:
```suggestion
* Fix Doxygen generation by @BewareMyPower in https://github.com/apache/pulsar-client-cpp/pull/103
```

File: site2/website-next/release-notes/versioned/client-cpp-3.1.0.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+* [feat] Support WaitForExclusive producer access mode. by @shibd in https://github.com/apache/pulsar-client-cpp/pull/109
+* [fix] Fix wrong behavior when removing the chunkedMessageCtx by @RobertIndie in https://github.com/apache/pulsar-client-cpp/pull/110
+* [Improve][Build] Use pulsar client as a submodule for C++ projects by @fregate in https://github.com/apache/pulsar-client-cpp/pull/115
+* Fix getLastMessageId method dead recursion. by @shibd in https://github.com/apache/pulsar-client-cpp/pull/117
+* [feat] Support consumer seek by timestamp and reader seek for C Api by @RobertIndie in https://github.com/apache/pulsar-client-cpp/pull/118
+
+## New Contributors
+* @fregate made their first contribution in https://github.com/apache/pulsar-client-cpp/pull/115
+
+**Full Changelog**: https://github.com/apache/pulsar-client-cpp/commits/v3.1.0
```
Comment (Anonymitaet) [inline site2/website-next/release-notes/versioned/client-cpp-3.1.0.md:41]: Shall we move line 41 to line 38 since they all talk about "changes"?

**Review by Anonymitaet — APPROVED** [review]

File: site2/website-next/release-notes/versioned/client-cpp-3.1.0.md (hunk 2 above)
Comment (RobertIndie) [reply site2/website-next/release-notes/versioned/client-cpp-3.1.0.md]: Good idea.

**Comment by RobertIndie** [comment]
The vote has passed: https://lists.apache.org/thread/by3hm2z677qfj3v7tqf719xd33pvoyxd. Merging this RP.

**Comment by tisonkun** [comment]
@RobertIndie to generate API docs for C++ client 3.1.0, you may run:

```shell
cd pulsar-site
git checkout -b gen-cpp-apidoc
cd tools/pytools
poetry install
poetry run bin/cpp-apidoc-generator.py 3.1.0 
```

Read [pytool's README](https://github.com/apache/pulsar-site/blob/main/tools/pytools/README.md) for more about setups, and throw issues to improve the release process if anything confusing.

**Comment by RobertIndie** [comment]
@tisonkun Hi, the release process misses this step. I have opened a PR to generate the c++ doc: https://github.com/apache/pulsar-site/pull/336.
And also added `Generate C++ Client docs` step to the release process: https://github.com/apache/pulsar-client-cpp/wiki/Committer-Release-Guide#generate-c-client-docs

**Pull request merged**